### ISSUE TITLE
Bug: Persona 4 Golden freezes with Reloaded II 1.21.0

### ISSUE THREAD

**Issue by viiibe** [Reloaded-Project/Reloaded-II#141]
**Describe the Bug**
Persona 4 Golden freezes on launch with Reloaded II v1.21.0

**To Reproduce**
Steps to reproduce the behaviour:
Launch Persona 4 Golden with PC Mod Loader enabled.

**Expected Behaviour**
Game launches and works.

**Additional Context**
I came here from CEP Discord and did all the necessary troubleshooting to rule out any other issues. Prerequisites installed. Windows 11 version Insider Preview 22623.870 ni_release.

1.20.8 works just fine.

**Comment by viiibe** [comment]
The logs show no abnormalities whatsoever.

```

log: ```
[22:53:11] [Reloaded] Create Loader | Time: 78ms
[22:53:11] [Reloaded] Current Process Location: D:\Games\Steam\steamapps\common\Persona 4 Golden\P4G.exe
[22:53:11] [Reloaded] Loading Assembly Metadata for Inter Mod Communication, Determining Unload Support etc.: Complete 25ms
[22:53:11] [Reloaded] Prepared Mod: Reloaded.Memory.SigScan.ReloadedII: Complete 3ms
[22:53:11] [Reloaded] Prepared Mod: p4gpc.tinyfixes: Complete 3ms
[22:53:11] [Reloaded] Prepared Mod: p4gpc.inaba: Complete 3ms
[22:53:11] [Reloaded] Prepared Mod: reloaded.sharedlib.hooks: Complete 3ms
[22:53:11] [Reloaded] Prepared Mod: p4gpc.modloader: Complete 8ms
[22:53:11] [Reloaded] Prepare All Mods (Total): Complete 9ms
[22:53:11] [Reloaded] Initialized Mod: reloaded.sharedlib.hooks: Complete 0ms
[22:53:11] [R-II SteamHook] SteamAPI_RestartAppIfNecessary hooked successfully.
[22:53:11] [R-II SteamHook] SteamAPI_IsSteamRunning hooked successfully.
[22:53:11] [R-II SteamHook] Found Steam Library Entry with Id 1113000. Dropping steam_appid.txt.
[22:53:11] [Reloaded] Setting Up Hooks (Async) | Time: 160ms
[22:53:11] [modloader] Persona 4 Golden (Steam) Mod loader by TGE (2020) v1.4.3
[22:53:11] [Reloaded] Initialized Mod: p4gpc.modloader: Complete 296ms
[22:53:16] [Reloaded] Initialized Mod: p4gpc.tinyfixes: Complete 4722ms
[22:53:16] [Reloaded] Initialized Mod: Reloaded.Memory.SigScan.ReloadedII: Complete 2ms
[22:53:16] [Reloaded] Initialized Mod: p4gpc.inaba: Complete 6ms
[22:53:16] [Reloaded] Initialized All Mods (Total): Complete 5028ms
[22:53:16] [Reloaded] Loading Mods (Total) | Time: 5101ms
[22:53:16] [Reloaded] Total Loader Initialization Time: 5182ms
```

```

**Comment by Sewer56** [comment]
Is there a crash dump available?
These days when there's a crash, Reloaded should produce a crash dump (.dmp) file when possible alongside the log. Usually a window informs you of the crash occurring.

Aside the move from .NET 5 to .NET 7 there aren't really any changes in loader logic between the two versions.

**Comment by viiibe** [comment]
It did not inform me of the crash to my knowledge

**Comment by Sewer56** [comment]
Can you grab event viewer log as per https://github.com/Reloaded-Project/Reloaded-II/issues/136#issuecomment-1267457643 ?
Maybe there's something in there.

**Comment by viiibe** [comment]
like this?

[error.zip](https://github.com/Reloaded-Project/Reloaded-II/files/9842699/error.zip)

**Comment by Sewer56** [comment]
Yeah, except find one for P4G.exe with source that says `.NET Runtime`

**Comment by viiibe** [comment]
also application hang when starting. i do not see any other errors, no recent ones tagged .NET Runtime

[applicationhang.zip](https://github.com/Reloaded-Project/Reloaded-II/files/9842721/applicationhang.zip)

**Comment by Sewer56** [comment]
Well, this is problematic... I've no idea for now; but the fact it just hangs and stops responding is weird. There wasn't any loader code changes in 1.21.0 aside from a bugfix related to the crash handler.

**Comment by viiibe** [comment]
Yeah, as said, 1.20.8 works fine. I did make sure to get the prerequisites

**Comment by viiibe** [comment]
1.21.1 still doesn’t work. No logs, everything is as if things were normal except it makes the game freeze nonetheless.

**Comment by Sewer56** [comment]
Yeah 1.21.1 is just a hotfix for misc. bugs; right now I really have no idea what's going on in your case.

**Comment by Sewer56** [comment]
Checking if it happens without mods, or trying to narrow it down to a mod might help.

**Comment by viiibe** [comment]
Seems like P4G PC Mod Loader is the culprit actually. I tested just with it and it froze, but not with two other mods that came with initial install

**Comment by Sewer56** [comment]
Oh boy I do not look forward to this; we can try some random stuff soon though :v

**Comment by viiibe** [comment]
Issue fixed as of 1.22.7 for me
